src/components/AddToCartForm/AddToCartForm.jsx

```diff
@@ -1,50 +1,35 @@
 import { useState } from "react";
+import { useCart } from "../../contexts/CartContext";
+import QuantitySelector from "../QuantitySelector/QuantitySelector";
 import {
   AddToCartButton,
   AddToCartContainer,
   AddToCartInput,
-  InputButton,
-  InputControls,
   InputLabel,
-  InputText,
 } from "./AddToCartForm.styled";
 
-export default function AddToCartForm() {
-  const [quantity, setQuantity] = useState(1);
+export default function AddToCartForm({ product }) {
+  const [quantity, setQuantity] = useState();
+  const { cart, addItem } = useCart();
 
-  function handleQuantityChange(e) {
-    let newValue = e.target.value;
-    if (isNaN(newValue) || newValue < 1) {
-      newValue = 1;
-    }
-    setQuantity(parseInt(newValue));
-  }
+  const cartItem = cart.find(({ item }) => item.id === product.id);
+  const disabledButton = cartItem?.quantity >= cartItem?.item.data.stock;
 
   return (
     <AddToCartContainer>
       <AddToCartInput>
         <InputLabel htmlFor="item_quantity">Qty.:</InputLabel>
```

Comment:
we should probably export the `name` from QuantitySelector

Reply:
You're right, I decided to pass the name and id as a prop 46a78a9